ansible_collections/ds/ansible_ds/plugins/module_utils/dsentities.py

```diff
@@ -0,0 +1,904 @@
+            del dictCopy[p]
+        for o in self.OPTIONS:
+            if o.name in dictCopy:
+                del dictCopy[o.name]
+        if len(dictCopy) > 0:
+            raise yaml.YAMLError(f"Unexpected  parameters {dictCopy.keys()} in {self.getClass()} object {self}")
+            
+    def __repr__(self):
+        return f"{self.__class__.__name__}(variables={self.__dict__})"
+        return f"{self.__class__.__name__}(name={self.name}, variables={self.__dict__})"
+
+    def getFacts(self):
+        ### populate the object (should be implemented in subclasses)
+        assert False
+
+    def findFact(self, facts):
+        if not facts:
+            return facts
+        if self.getClass() == facts.getClass() and self.name == facts.name:
+            return facts
+        for var in facts.CHILDREN:
+            list = facts.__dict__[var]
+            for c in list:
+                if self.getClass() == c.getClass() and self.name == c.name:
+                    return c
+        facts = globals()[self.getClass()](self.name)
+        facts.state = "absent"
+        return facts
+        
+    def getType(self):
+        return self.getClass().replace("YAML","")
+
+    def getAllActions(self, facts):
+        actions = []
+        for option in self.OPTIONS:
+            vfrom = getattr(facts, option.name, None)
+            vto = getattr(self, option.name, None)
+            for action in option._get_action(self, facts, vfrom, vto):
+                actions.append(action)
+        return sorted(actions, key = lambda x : x.getPrio())
+
+    def update(self, facts=None, args=None):
+        if not facts:
+            facts = YAMLHost()
+            facts.getFacts()
+        facts = self.findFact(facts)
+
+        log.debug(f"Updating instance {self.name}  with facts {facts.name}")
+
+        actions = self.getAllActions(facts)
+        for action in actions:
+            if self._isDeleted is True:
+                return
+            if action.vfrom == action.vto:
+                continue
+            msg = action.perform(OptionAction.DESC)
+            if (msg):
+                log.info(msg)
+            if getattr(args, "no_actions", False):
+                continue
+            action.perform(OptionAction.UPDATE)
+        inst = self.getYAMLInstance()
+        if inst:
+            inst.applyMods(self._cfgMods)
+            dseMods = getattr(self, "dseMods", None)
+            if dseMods:
+                inst.applyMods(dseMods)
+
+        for var in self.CHILDREN:
+            list = self.__dict__[var]
+            for c in list:
+                c.update(facts, args)
+
+    def addModifier(self, dn, type, attr, val):
+        dict = self._cfgMods
+        DiffResult.addModifier(dict, dn, type, attr, val)
+
+
+class YAMLHost(MyYAMLObject):
+    yaml_tag = u'!ds389Host'
+
+    OPTIONS = (
+        SpecialOption('prefix', 1, "389 Directory Service non standard installation path" ),
```

Comment:
Do I understand correctly that currently, the only way to test/run the setup is to have 389-ds-base installed in a PREFIX install?

Reply:
No, the parameter is optional. It should also  work with a standard installation from pkg. That said, I did not tested it yet, so there may be bugs ! -;)